src/Zf2Logger/Factory/Zf2Logger.php

```diff
@@ -65,7 +69,20 @@ private function writerCollection(array $config)
      */
     private function writerAdapter(array $writer)
     {
-        $writerAdapter = new $writer['adapter']($writer['options']['output']);
+        $sm = $this->getServiceLocator();
+        if ($sm->has($writer['adapter'])) {
+            $writerAdapter = $sm->get($writer['adapter']);
+            $methods = get_class_methods(get_class($writerAdapter));
+            if (
+                !empty($writer['options']) &&
+                in_array('setOptions', $methods)
```

Comment:
line 75 & 78 could be replaced with `method_exists($writerAdapter, 'setOptions')` http://php.net/manual/en/function.method-exists.php